logicrunner/handle_abandonedrequestsnotification.go

```diff
@@ -0,0 +1,101 @@
+//
+// Copyright 2019 Insolar Technologies GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package logicrunner
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+
+	"github.com/insolar/insolar/insolar/flow"
+	"github.com/insolar/insolar/insolar/flow/bus"
+	"github.com/insolar/insolar/insolar/message"
+	"github.com/insolar/insolar/insolar/reply"
+	"github.com/insolar/insolar/instrumentation/inslogger"
+	"github.com/insolar/insolar/instrumentation/instracer"
+)
+
+type initializeAbandonedRequestsNotificationExecutionState struct {
+	LR  *LogicRunner
+	msg *message.AbandonedRequestsNotification
+}
+
+// Proceed initializes or sets LedgerHasMoreRequests to right value
+func (p *initializeAbandonedRequestsNotificationExecutionState) Proceed(ctx context.Context) error {
+	ref := *p.msg.DefaultTarget()
+
+	state := p.LR.UpsertObjectState(ref)
+
+	state.Lock()
+	if state.ExecutionState == nil {
+		state.ExecutionState = &ExecutionState{
+			Ref:                   ref,
+			Queue:                 make([]ExecutionQueueElement, 0),
+			pending:               message.InPending,
+			PendingConfirmed:      false,
+			LedgerHasMoreRequests: true,
+		}
+	} else {
+		executionState := state.ExecutionState
+		executionState.Lock()
+		executionState.LedgerHasMoreRequests = true
+		executionState.Unlock()
+
+	}
+	state.Unlock()
+
+	return nil
+}
+
+// ---------------
```

Comment:
please don't leave such kind of comments.

Reply:
fixed